components/camel-seda/src/main/java/org/apache/camel/component/seda/SedaEndpoint.java

```diff
@@ -100,6 +100,8 @@ public class SedaEndpoint extends DefaultEndpoint implements AsyncEndpoint, Brow
     private boolean discardIfNoConsumers;
 
     private BlockingQueueFactory<Exchange> queueFactory;
+    private String key;
+    private QueueReference ref;
```

Comment:
Only `ref` seems to be needed here. Moreover, please note that it should be `volatile` otherwise you have unfortunately no guarantee that its value will be visible from a thread executed by a different core

Reply:
I've just checked the code. We do have a few occasions where this could happen. Let's make it `volatile` here. Thanks @essobedo!